### PR TITLE
The common scripts of PR git diff changes.

### PR DESCRIPTION
**Pull request by sima-zhu**
Have a common script for git diff changes.
Test proposal:
PR triggered(Compliance credscan): https://github.com/Azure/azure-sdk-for-js/pull/20643
Manually triggered(Build print diff): https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1407323&view=results
Test on Add, Update, Delete, Rename cases(Compliance credscan): https://github.com/Azure/azure-sdk-for-js/pull/20643
Test on sparse-checkout cases(Build print diff): https://github.com/Azure/azure-sdk-for-java/pull/27335

image
The git diff changes will add the file outside of the sparse checkout

### CODE REVIEW

File: eng/common/scripts/git-diff-changes.ps1 [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+  .PARAMETER TargetCommittish
+  The branch committish PR targets to merge into.
+#>
+[CmdletBinding()]
+param (
+  [string] $SourceCommittish = ${env:BUILD_SOURCEVERSION},
+  [string] $TargetCommittish = ("origin/${env:SYSTEM_PULLREQUEST_TARGETBRANCH}" -replace "refs/heads/")
+)
+# Git PR diff: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-comparing-branches-in-pull-requests#three-dot-and-two-dot-git-diff-comparisons
```
Comment (benbp) [inline eng/common/scripts/git-diff-changes.ps1:20]: I imagine our link checker won't like the `/en/` segment in there

**Review by chidozieononiwu — APPROVED** [review]
Works for the eng/common workflow enforcer case

File: eng/common/scripts/git-diff-changes.ps1 [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+Write-Host "git -c core.quotepath=off -c i18n.logoutputencoding=utf-8 diff $TargetCommittish...$SourceCommittish --name-only --diff-filter=d"
+$changedFiles = (git -c core.quotepath=off -c i18n.logoutputencoding=utf-8 diff "$TargetCommittish...$SourceCommittish"  --name-only --diff-filter=d)
+if(!$changedFiles) {
+Write-Host "No changed files in git diff between $TargetCommittish and $SourceCommittish"
+}
+foreach ($file in $changedFiles) {
+Write-Host "The diff file is: $file"
+}
```
Comment (benbp) [inline eng/common/scripts/git-diff-changes.ps1:35]:
```suggestion
if(!$changedFiles) {
    Write-Host "No changed files in git diff between $TargetCommittish and $SourceCommittish"
}
foreach ($file in $changedFiles) {
    Write-Host "The diff file is: $file"
}
```

**Review by benbp — APPROVED** [review]

**Comment by kushagraThapar** [comment]
@sima-zhu / @benbp - seems like the above azure-sdk-tools - sync - eng-common CI is stuck / or waiting for some approval. 
Can we please merge this PR soon, so that other Cosmos CI PRs can be merged, this is currently blocking our Cosmos CI, and all development / release process. Thanks!

File: eng/common/scripts/git-diff-changes.ps1 [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# Git PR diff: https://docs.github.com/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-comparing-branches-in-pull-requests#three-dot-and-two-dot-git-diff-comparisons
+Write-Host "git -c core.quotepath=off -c i18n.logoutputencoding=utf-8 diff $TargetCommittish...$SourceCommittish --name-only --diff-filter=d"
+$changedFiles = (git -c core.quotepath=off -c i18n.logoutputencoding=utf-8 diff "$TargetCommittish...$SourceCommittish"  --name-only --diff-filter=d)
+if(!$changedFiles) {
+    Write-Host "No changed files in git diff between $TargetCommittish and $SourceCommittish"
+}
+Write-Host "Here are the diff files:"
+foreach ($file in $changedFiles) {
+    Write-Host "    $file"
+}
```
Comment (weshaggard) [inline eng/common/scripts/git-diff-changes.ps1:35]: We should put this block inside the else from above so we don't end up with extra logging in the empty case.

File: eng/common/scripts/git-diff-changes.ps1 [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  Definition of committish: https://git-scm.com/docs/gitglossary#Documentation/gitglossary.txt-aiddefcommit-ishacommit-ishalsocommittish
+
+  .PARAMETER TargetCommittish
+  The branch committish PR targets to merge into.
+#>
+[CmdletBinding()]
+param (
+  [string] $SourceCommittish = "${env:BUILD_SOURCEVERSION}",
+  [string] $TargetCommittish = ("origin/${env:SYSTEM_PULLREQUEST_TARGETBRANCH}" -replace "refs/heads/")
+)
```
Comment (weshaggard) [inline eng/common/scripts/git-diff-changes.ps1:19]: We probably also need an optional parameter to pass in a path filter. (i.e. for things like eng/common).

**Pull request merged**